src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -411,32 +411,31 @@
 
 (rf/reg-event-fx
  :wallet.swap/approve
- (fn [{:keys [db]}]
-   (let [last-request-uuid (get-in db [:wallet :ui :swap :last-request-uuid])
-         max-slippage      (get-in db [:wallet :ui :swap :max-slippage])]
-     {:fx [[:dispatch
-            [:wallet/build-transactions-from-route
-             {:request-uuid last-request-uuid
-              :slippage     max-slippage}]]
-           [:dispatch
-            [:wallet.swap/set-sign-transactions-callback-fx
-             [:dispatch [:open-modal :screen/wallet.swap-set-spending-cap]]]]]})))
+ (fn [_]
+   {:fx [[:dispatch [:open-modal :screen/wallet.swap-set-spending-cap]]]}))
```

Comment:
Given we are just dispatching another event in the effects, maybe this event is not necessary anymore, and we could dispatch `:open-modal` directly instead.

Reply:
Good point, applied .